public/app/plugins/panel/barchart/bars.ts

```diff
@@ -352,13 +353,24 @@ export function getConfig(opts: BarsOptions, theme: GrafanaTheme2) {
         let middleShift = isXHorizontal ? 0 : -Math.round(MIDDLE_BASELINE_SHIFT * fontSize);
         let value = rawValue(seriesIdx, dataIdx);
 
-        if (value != null) {
+        let xFormedValue = value;
+
+        if (opts.negY?.[seriesIdx] && xFormedValue != null) {
+          xFormedValue *= -1;
+        }
+
+        if (xFormedValue != null) {
           // Calculate final co-ordinates for text position
           const x =
-            u.bbox.left + (isXHorizontal ? lft + wid / 2 : value < 0 ? lft - labelOffset : lft + wid + labelOffset);
+            u.bbox.left +
+            (isXHorizontal ? lft + wid / 2 : xFormedValue < 0 ? lft - labelOffset : lft + wid + labelOffset);
           const y =
             u.bbox.top +
-            (isXHorizontal ? (value < 0 ? top + hgt + labelOffset : top - labelOffset) : top + hgt / 2 - middleShift);
+            (isXHorizontal
+              ? xFormedValue < 0
+                ? top + hgt + labelOffset
+                : top - labelOffset
+              : top + hgt / 2 - middleShift);
```

Comment:
i hate this Prettier auto-formatting in a pre-commit hook. the only change here is renaming `value` -> `xFormedValue`.

same story for the diff below.